packages/aws-cdk-lib/aws-events-targets/lib/ecs-task.ts

```diff
@@ -50,6 +50,54 @@ export interface EcsTaskProps extends TargetBaseProps {
    */
   readonly containerOverrides?: ContainerOverride[];
 
+  /**
+   * The CPU override for the task.
+   *
+   * @default - The task definition's CPU value
+   */
+  readonly cpu?: string;
+
+  /**
+   * The ephemeral storage setting override for the task.
+   *
+   * NOTE: This parameter is only supported for tasks hosted on Fargate that use the following platform versions:
+   *  - Linux platform version 1.4.0 or later.
+   *  - Windows platform version 1.0.0 or later.
+   *
+   * @default - The task definition's ephemeral storage value
+   */
+  readonly ephemeralStorage?: EphemeralStorageOverride;
```

Comment:
Nit: since there is only one property in the `EphemeralStorageOverride` interface would it be better to just support that there? Something like `ephemeralStorageSizeInGiB: number`

Reply:
I copied the upstream CFN object, since it's an object: https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_EphemeralStorage.html

Since they made it an object in CFN, my thinking is that there might be additional properties added later on, so I followed that pattern here.

However, if you'd prefer the L2 implementation to hide this detail, I can make it a single property. Just let me know.

Reply:
I'd prefer to keep the interface as flat as possible. While creating a task in the console I see there are at least 3 different settings users can configure for Ephemeral storage so presumably these will be props under this interface when they're supported in CloudFormation.
<img width="945" alt="Screenshot 2025-02-10 at 4 29 10 PM" src="https://github.com/user-attachments/assets/45985556-27e0-4801-b4c9-682778b570c6" />

However, I don't think these props will be connected (they'll all be optional) so IMO this should be a single property and if the other settings I mentioned are added in CFN then they should be individual props as well. Curious about your thoughts on this?

Reply:
talked it over with the team and they point out that in this specific case we should keep it a interface since the console already has at least 3 options those will likely become cfn properties soon so its better to future proof